lib/mayaUsd/fileio/writeJobContext.cpp

```diff
@@ -385,6 +385,39 @@ bool UsdMayaWriteJobContext::_NeedToTraverse(const MDagPath& curDag) const
         // If we're not going to create a prim at curDag, then we do not need to
         // traverse.
         return false;
+    } else {
+        // In addition to check for primless, we check for user selection of export types
+        if (mArgs.excludeExportTypes.empty()) {
+            return true;
+        }
+
+        MDagPath shapeDagPath = curDag;
+        if (mArgs.mergeTransformAndShape) {
+            // if we're merging transforms, then we need to look at the shape.
+            shapeDagPath.extendToShape();
+        }
+
+        MStatus                 status;
+        MObject                 obj = shapeDagPath.node();
+        const MFnDependencyNode depFn(obj, &status);
+        if (!status) {
+            return false;
+        }
+
+        const std::string mayaTypeName(depFn.typeName().asChar());
+
+        if (mArgs.excludeExportTypes.count(TfToken("Meshes")) != 0) {
```

Comment:
Were  the name of types part of the design? I ask because I find that using "Meshes" instead of "mesh" for a type name feels wrong. The type should not be plural.

In any cases, the string comparison should be case-insensitive.